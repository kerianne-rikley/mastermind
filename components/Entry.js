import * as React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

const colourMap = {0:'#1e90ff', 1: '#ee82ee', 2: '#dc143c', 3: '#ff8c00', 4: '#ffd700', 5: '#228b22'};

export default class Entry extends React.Component{
    state = {value: 0};

    updateState = () => {
        this.setState({value: (this.state.value+1)%6})
    }

    render(){
        return (
            <View style={styles.container}>
                <Button
                    title={this.state.value}
                    color = {colourMap[this.state.value]}
                    onPress = {() => {this.updateState()}}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
    },
    titleCard: {
        fontSize: 30,
        fontWeight: 'bold',
        marginVertical: 10,
    },
});
