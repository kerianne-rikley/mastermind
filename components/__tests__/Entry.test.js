import React from "react";
import TestRenderer from "react-test-renderer";
import { shallow } from "enzyme";

import Entry from "../Entry";

describe("<Entry />", () => {

  const handlePressMock = jest.fn(); // Just an empty function

  it("renders correctly", () => {
    const renderedEntry = TestRenderer.create(
      <Entry id={0} choice="0" handlePress={handlePressMock} />).toJSON();
    expect(renderedEntry).toMatchSnapshot();
  });

  describe("entry button", () => {
    // I'm using enzyme here instead of testing-library because it
    // supports getting properties better

    let wrapper;
    let entryButton;

    const fakeId = 0;
    const fakeChoice = "3";

    beforeAll(() => {
      wrapper = shallow(
        <Entry id={fakeId} handlePress={handlePressMock} choice={fakeChoice} />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it("exists", () => {
      entryButton = wrapper.find(`#entry-button-${fakeId}`).getElement();
      expect(entryButton).toBeTruthy();
    });

    it("executes the handlePress function when clicked", () => {
      entryButton = wrapper.find(`#entry-button-${fakeId}`).getElement();
      entryButton.props.onPress();
      expect(handlePressMock).toHaveBeenCalledTimes(1);
      // Check the arguments passed to mock function 
      expect(handlePressMock).toHaveBeenCalledWith(fakeId); 
    });

    it("displays the choice as a title", () => {
      entryButton = wrapper.find(`#entry-button-${fakeId}`).getElement();
      expect(entryButton.props.title).toEqual(fakeChoice);
    });

    it("displays the choice color", () => {
      entryButton = wrapper.find(`#entry-button-${fakeId}`).getElement();
      expect(entryButton.props.color).toEqual("#ff8c00");
    });

    describe("displays choice color", () => {
      const colors = [
        "#316EFF", // "#1e90ff",
        "#ee82ee",
        "#dc143c",
        "#ff8c00",
        "#ffd700",
        "#228b22"
      ];

      colors.forEach((color, index) => {
        it(`displays ${color} for ${index}`, () => {
          wrapper = shallow(
            <Entry id={fakeId} choice={index} />);  
          entryButton = wrapper.find(`#entry-button-${fakeId}`).getElement();
          expect(entryButton.props.color).toEqual(color);
        });
      });
    });
  });
});
