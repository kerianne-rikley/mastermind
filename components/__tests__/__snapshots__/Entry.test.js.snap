// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entry /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": 40,
      "margin": 3,
      "padding": 2,
      "width": 40,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#316EFF",
            },
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
